File: AKW/frontend/src/api/roadmap.ts
import client from './client'

export interface RoadmapItem {
  id: string
  project_id: string
  version_id: string | null
  priority: 'p1' | 'p2' | 'p3' | 'p4'
  category: string
  title: string
  description: string | null
  size: 'XS' | 'S' | 'M' | 'L' | 'XL'
  status: string
  sort_order: number
  created_at: string
  updated_at: string
}

export type RoadmapItemCreate = Pick<RoadmapItem, 'priority' | 'category' | 'title' | 'size' | 'status'> & {
  description?: string
  sort_order?: number
  version_id?: string | null
}

export type RoadmapItemUpdate = Partial<Omit<RoadmapItem, 'id' | 'project_id' | 'created_at' | 'updated_at'>>

export const roadmapApi = {
  list: (projectId: string, versionId?: string | null) => {
    const params = versionId ? { version_id: versionId } : {}
    return client.get<RoadmapItem[]>(`/projects/${projectId}/roadmap`, { params }).then(r => r.data)
  },
  create: (projectId: string, data: RoadmapItemCreate) =>
    client.post<RoadmapItem>(`/projects/${projectId}/roadmap`, data).then(r => r.data),
  update: (id: string, data: RoadmapItemUpdate) =>
    client.patch<RoadmapItem>(`/roadmap/${id}`, data).then(r => r.data),
  delete: (id: string) =>
    client.delete(`/roadmap/${id}`),
}

File: AKW/backend/app/schemas/roadmap.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict


class RoadmapItemCreate(BaseModel):
    priority: Literal["p1", "p2", "p3", "p4"] = "p3"
    category: str = "공통"
    title: str
    description: str | None = None
    size: Literal["XS", "S", "M", "L", "XL"] = "M"
    status: str = "todo"
    sort_order: int = 0
    version_id: uuid.UUID | None = None


class RoadmapItemUpdate(BaseModel):
    priority: Literal["p1", "p2", "p3", "p4"] | None = None
    category: str | None = None
    title: str | None = None
    description: str | None = None
    size: Literal["XS", "S", "M", "L", "XL"] | None = None
    status: str | None = None
    sort_order: int | None = None
    version_id: uuid.UUID | None = None


class RoadmapItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    version_id: uuid.UUID | None
    priority: str
    category: str
    title: str
    description: str | None
    size: str
    status: str
    sort_order: int
    created_at: datetime
    updated_at: datetime
